engine/src/main/java/org/camunda/bpm/engine/impl/NativeHistoricVariableInstanceQueryImpl.java

```diff
@@ -0,0 +1,70 @@
+/* Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.impl;
+
+import org.camunda.bpm.engine.history.HistoricVariableInstance;
+import org.camunda.bpm.engine.history.NativeHistoricVariableInstanceQuery;
+import org.camunda.bpm.engine.impl.cmd.CommandLogger;
+import org.camunda.bpm.engine.impl.interceptor.CommandContext;
+import org.camunda.bpm.engine.impl.interceptor.CommandExecutor;
+import org.camunda.bpm.engine.impl.persistence.entity.HistoricVariableInstanceEntity;
+
+import java.util.List;
+import java.util.Map;
+
+
+public class NativeHistoricVariableInstanceQueryImpl extends AbstractNativeQuery<NativeHistoricVariableInstanceQuery, HistoricVariableInstance>
+        implements NativeHistoricVariableInstanceQuery {
+
+  private final static CommandLogger LOG = ProcessEngineLogger.CMD_LOGGER;
+
+  private static final long serialVersionUID = 1L;
+
+  public NativeHistoricVariableInstanceQueryImpl(CommandContext commandContext) {
+    super(commandContext);
+  }
+
+  public NativeHistoricVariableInstanceQueryImpl(CommandExecutor commandExecutor) {
+    super(commandExecutor);
+  }
+
+
+  //results ////////////////////////////////////////////////////////////////
+
+  public List<HistoricVariableInstance> executeList(CommandContext commandContext, Map<String, Object> parameterMap, int firstResult, int maxResults) {
+    List<HistoricVariableInstance> historicVariableInstances = commandContext
+            .getHistoricVariableInstanceManager()
+            .findHistoricVariableInstancesByNativeQuery(parameterMap, firstResult, maxResults);
+
+    if (historicVariableInstances!=null) {
+      for (HistoricVariableInstance historicVariableInstance: historicVariableInstances) {
+
+        HistoricVariableInstanceEntity variableInstanceEntity = (HistoricVariableInstanceEntity) historicVariableInstance;
+          try {
+            variableInstanceEntity.getTypedValue(true);
```

Comment:
Please consider to have configurable field for this like isCustomObjectDeserializationEnabled in HistoricVariableInstanceQueryImpl and add test cases for it.